mlflow/models/evaluation/default_evaluator.py

```diff
@@ -540,9 +544,10 @@ def _log_model_explainability(self):
             "explainability_nsamples", _DEFAULT_SAMPLE_ROWS_FOR_SHAP
         )
 
-        truncated_feature_names = [truncate_str_from_middle(f, 20) for f in self.feature_names]
+        truncated_feature_names = [truncate_str_from_middle(str(f), 20) for f in self.feature_names]
```

Comment:
https://github.com/mlflow/mlflow/blob/9c2010c1bf4466ab6f78532411d945d79069c7ba/mlflow/models/evaluation/base.py#L363-L368

Can we stringify feature names here?

cc @WeichenXu123